app/services/shift_assignment_service.rb

```diff
@@ -0,0 +1,49 @@
+class ShiftAssignmentService
+
+  attr_reader :deliverer_id, :shift_id, :errors, :success
+
+  def initialize(deliverer_id, shift_id)
+    @deliverer_id = deliverer_id
+    @shift_id = shift_id
+    @errors = []
+    @success = []
+  end
+
+  def perform
+
+    # Terminate when variables are not found
+    if @deliverer_id.nil? || @shift_id.nil?
+      @errors << "Please create some Deliverers and Shifts first."
+      return false
+    end
+
+    # Check max count for Shift
+    shift = Shift.find(shift_id)
+    if shift.max_count == shift.deliverers.count
```

Comment:
it's safer to check using `<=`, since deliverers could have been assigned by some other way

Reply:
Ok I will take note of that